src/main/java/com/minecraftabnormals/upgrade_aquatic/common/entities/jellyfish/helper/JellyfishSizeMap.java

```diff
@@ -0,0 +1,106 @@
+package com.minecraftabnormals.upgrade_aquatic.common.entities.jellyfish.helper;
+
+import com.minecraftabnormals.abnormals_core.core.AbnormalsCore;
+import org.apache.logging.log4j.Level;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.TreeMap;
+
+public class JellyfishSizeMap {
+    protected final TreeMap<Float, Integer> sizeMap;
+    private final int totalWeight;
+
+    public JellyfishSizeMap(TreeMap<Float, Integer> sizeMap) {
+        this.sizeMap = sizeMap;
+        int totalWeight = 0;
+        for (int weight : this.sizeMap.values())
+            totalWeight += weight;
+        this.totalWeight = totalWeight;
+        if(this.sizeMap.isEmpty()) AbnormalsCore.LOGGER.throwing(Level.ERROR, new IllegalArgumentException("[Upgrade Aquatic] Inputted size map for JellyfishSizeMap cannot be empty!"));
```

Comment:
Just use `AbnormalsCore.LOGGER.error()`.
There's also no need to add `[Upgrade Aquatic]` since it'll print out where the error is coming from.
```suggestion
        if (this.sizeMap.isEmpty()) AbnormalsCore.LOGGER.error("Inputted size map cannot be empty!");
```

Reply:
Just throwing an `IllegalArgumentException` here with the message `"Inputted size map cannot be empty!"` would be totally fine as well.